x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/ESNativeRealmMigrateTool.java

```diff
@@ -363,9 +363,16 @@ static Logger getTerminalLogger(final Terminal terminal) {
         final Logger logger = ESLoggerFactory.getLogger(ESNativeRealmMigrateTool.class);
         Loggers.setLevel(logger, Level.ALL);
 
+        final LoggerContext ctx = (LoggerContext) LogManager.getContext(false);
+        final Configuration config = ctx.getConfiguration();
+
         // create appender
         final Appender appender = new AbstractAppender(ESNativeRealmMigrateTool.class.getName(), null,
-                PatternLayout.newBuilder().withPattern("%m").build()) {
+                PatternLayout.newBuilder()
+                    // Specify the configuration so log4j doesn't re-initialize
+                    .withConfiguration(config)
```

Comment:
Without this the test fails with a security error because log4j tries to re-initialize itself which calls `getClassloader` when there are log4j plugins which fails because of security permissions. Reusing the configuration here skips all that.